src/manager.cpp

```diff
@@ -334,6 +337,19 @@ sdbusplus::async::task<bool>
     constexpr size_t rsyncVanishedExitCode = 24;
     const fs::path currentSrcPath = srcPath.empty() ? dataSyncCfg._path
                                                     : srcPath;
+
+    if (retryCount == 0)
+    {
+        if (dataSyncCfg._syncInProgressPaths.find(currentSrcPath) !=
```

Comment:
You could use `contains` API instead of `find` and `end` APIs and these two `if` block could be combined 

```
if (retryCount == 0 && dataSyncCfg._syncInProgressPaths.contains(currentSrcPath))
{
    lg2::debug("Skipping sync for [{SRC}] since its already in progress", "SRC", currentSrcPath);
    co_return false;
}
```